src/org/centos/contra/pipeline/Utils.groovy

```diff
@@ -386,6 +386,25 @@ def trackMessage(String messageID, int retryCount, def dataGrepperWebAddr=null)
     }
 }
 
+/**
+ * Allow to check custom service for presence of a message
+ *
+ * It is possible to use method pointer operator (.&) from groovy
+ * if you do not want to create new closure.
+ * http://docs.groovy-lang.org/latest/html/documentation/core-operators.html#method-pointer-operator
+ *
+ * @param trackClosure closure for checking successful delivery of message.
+                       You need to throw exception (or to call error())
+                       in case of failure.
+ * @param retryCount number of times to keep trying.
+ * @return
+ */
+def trackMessage(Closure trackClosure, int retryCount) {
```

Comment:
trackMessage already exists in utils with two arguments. Can this be renamed to ```def trackClosure``` or something?

Reply:
Overloading of methods is natural part even in java. So there is not any conflict between old and new one. https://docs.oracle.com/javase/tutorial/java/javaOO/methods.html

IMHO `trackClosure` would be confusing here. Because we do not want to track "closure" but "message". Maybe `trackMessageWithClosure` ? Maybe even the name of 1st parameter is confusing.

But it is just a name so if you prefer `trackClosure` I can change that.

Reply:
Well, the other declaration of the function has two argument and one optional one, so I think having another declaration with two is quite confusing. I don't think there are any costs to using a different name and because of the similar number of arguments, I think there is a negative aspect to using the same name. Either of those names are fine by me, I just meant something not used

Reply:
I am not sure whether you notice example in java documentation [1]
but there is not any problem if there are even 3 functions with 1 parameter but different types.
```
public class DataArtist {
    ...
    public void draw(String s) {
        ...
    }
    public void draw(int i) {
        ...
    }
    public void draw(double f) {
        ...
    }
    public void draw(int i, double f) {
        ...
    }
}
```

That's the reason why it is called method overloading.

[1] https://docs.oracle.com/javase/tutorial/java/javaOO/methods.html

Reply:
I'll change that to `trackMessageWithClosure`

Reply:
I absolutely know about method overloading. I don't see what we gain from using it here, and we certainly lose readability because then people have to track types and find out what declaration of the function is being used. It takes people longer to figure the code out and the only gain is that the code is fancier. I do not think it should be used here.

Reply:
```trackMessageWithClosure``` works for me excellent thanks

Reply:
> I absolutely know about method overloading. I don't see what we gain from using it here, and we certainly lose readability because then people have to track types and find out what declaration of the function is being used. It takes people longer to figure the code out and the only gain is that the code is fancier. I do not think it should be used here.

We need to agree to disagree here
But It was already changed:-)